### PR TITLE
test:[HL-41] Splash view controller<> Unit Test🧪

### PR DESCRIPTION
**Pull request by djayyab**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: Healthy/Classes/Modules/Splash/SplashViewController.swift [hunk 1]
```diff
@@ -36,11 +36,10 @@ final class SplashViewController: UIViewController {
 
 // MARK: - Actions
 
-private extension SplashViewController {
+ extension SplashViewController {
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Splash/SplashViewController.swift]:
```suggestion
extension SplashViewController {
```

File: Healthy/Classes/Modules/Splash/SplashViewController.swift [hunk 2]
```diff
@@ -36,11 +36,10 @@ final class SplashViewController: UIViewController {
...
     @IBAction func didTapStartCooking(_ sender: Any) {
-        let viewModel = LoginViewModel()
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Splash/SplashViewController.swift]: We need the navigation logic. Currently, you can use both until we remove the navigation logic.

File: Healthy/Classes/Modules/Splash/SplashViewModelType.swift [hunk 3]
```diff
@@ -6,7 +6,9 @@ typealias SplashViewModelType = SplashViewModelInput & SplashViewModelOutput
 
 /// Splash ViewModel Input
 ///
-protocol SplashViewModelInput {}
+protocol SplashViewModelInput {
+    func performStartCooking()
+   }
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Splash/SplashViewModelType.swift]:
```suggestion
}
```

File: HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    }
+    }
+// Spy
+    private final class SplashViewModelMock: SplashViewModelType {
+    private(set)var performStartCookingCallCount: Int = .zero
+
+    func performStartCooking() {
+        performStartCookingCallCount += 1
+    }
+}
```
Comment (ahmdmhasn) [inline HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift]:
```suggestion
import XCTest
@testable import Healthy

final class SplashViewControllerTest: XCTestCase {
    
    // MARK: Properties
    
    private var viewController: SplashViewController!
    private var viewModelMock: SplashViewModelSpy!
    
    // MARK: Lifecycle
    
    override func setUpWithError() throws {
        viewModelMock = SplashViewModelSpy()
        viewController = SplashViewController(viewModel: viewModelMock)
    }
    
    // MARK: Tests
    
    func test_didTapStartCooking_shouldCallViewModelSplash() {
        // When
        viewController.didTapStartCooking(UIButton())
        
        // Then
        XCTAssertEqual(viewModelMock.performStartCookingCallCount, 1)
    }
}

// MARK: SplashViewModelSpy
private final class SplashViewModelSpy: SplashViewModelType {
    private(set)var performStartCookingCallCount: Int = .zero

    func performStartCooking() {
        performStartCookingCallCount += 1
    }
}
```

Please give attention to spaces and line spacings.

**Comment by ahmdmhasn** [comment]
Also, in the future, please avoid adding emojis to the branch name. You can still add it to the pull request title.

File: Healthy/Classes/Modules/Splash/SplashViewController.swift (hunk 2 above)
Comment (djayyab) [reply Healthy/Classes/Modules/Splash/SplashViewController.swift:41]: I forgot to remove it after I had deleted the navigation logic

Comment (djayyab) [reply Healthy/Classes/Modules/Splash/SplashViewController.swift:41]: Do we still need to put the navigation logic? I think the navigation was fixed in SceneDelegate by Mohamed recently

File: HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    func test_didTapStartCooking_shouldCallViewModelSplash() {
+        // When
+        viewController.didTapStartCooking(UIButton())
+
+        // Then
+        XCTAssertEqual(viewModelMock.performStartCookingCallCount, 1)
+    }
+}
+// MARK: SplashViewModelSpy
```
Comment (ahmdmhasn) [inline HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift]:
```suggestion
}

// MARK: SplashViewModelSpy
```

File: HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import XCTest
+@testable import Healthy
+final class SplashViewControllerTest: XCTestCase {
```
Comment (ahmdmhasn) [inline HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift]:
```suggestion
@testable import Healthy

final class SplashViewControllerTest: XCTestCase {
```

File: HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+    private var viewController: SplashViewController!
+    private var viewModelMock: SplashViewModelSpy!
+
+    // MARK: Lifecycle
+
+    override func setUpWithError() throws {
+        viewModelMock = SplashViewModelSpy()
+        viewController = SplashViewController(viewModel: viewModelMock)
+    }
```
Comment (ahmdmhasn) [inline HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift]:
```suggestion
    override func setUp() {
        super.setUp()
        
        viewModelMock = SplashViewModelSpy()
        viewController = SplashViewController(viewModel: viewModelMock)
    }
```

File: Healthy/Classes/Modules/Splash/SplashViewController.swift [hunk 8]
```diff
@@ -36,11 +36,12 @@ final class SplashViewController: UIViewController {
 
 // MARK: - Actions
 
-private extension SplashViewController {
+extension SplashViewController {
     @IBAction func didTapStartCooking(_ sender: Any) {
-        let viewModel = LoginViewModel()
-        let viewController = LoginViewController(viewModel: viewModel)
+        let loginviewModel = LoginViewModel()
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Splash/SplashViewController.swift]: Please revert the naming here. 
```suggestion
        let viewModel = LoginViewModel()
```

File: Healthy/Classes/Generated/UIImage.Generated.swift
Comment (ahmdmhasn) [inline Healthy/Classes/Generated/UIImage.Generated.swift:1]: Should be removed.

File: Healthy/Classes/Modules/Splash/SplashViewController.swift (hunk 8 above)
Comment (djayyab) [reply Healthy/Classes/Modules/Splash/SplashViewController.swift]: I have another variable called viewModel for a splash

Comment (ahmdmhasn) [reply Healthy/Classes/Modules/Splash/SplashViewController.swift]: But we need only one in the scope of the method. You can still access the outer view model with `self.viewModel`

File: Healthy/Classes/Generated/UIImage.Generated.swift
Comment (djayyab) [reply Healthy/Classes/Generated/UIImage.Generated.swift:1]: why should I remove it? it is not duplicating

Comment (ahmdmhasn) [reply Healthy/Classes/Generated/UIImage.Generated.swift:1]: It's not the same as cuurent generated assets file. 

Is should exist under /ResourcesGenerated/UIImages.Generated.swift

**Review by esraakhaled — APPROVED** [review]

File: HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+final class SplashViewControllerTest: XCTestCase {
+
+    // MARK: Properties
+
+    private var viewController: SplashViewController!
+    private var viewModelMock: SplashViewModelSpy!
+
+    // MARK: Lifecycle
+
+    override func setUpWithError() throws {
```
Comment (mhmdatallaa) [inline HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift]:
```suggestion
    override func setUp() {
```

File: HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        viewController.didTapStartCooking(UIButton())
+
+        // Then
+        XCTAssertEqual(viewModelMock.performStartCookingCallCount, 1)
+    }
+}
+
+// MARK: SplashViewModelSpy
+
+private final class SplashViewModelSpy: SplashViewModelType {
```
Comment (mhmdatallaa) [inline HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift]:
```suggestion
private final class SplashViewModelMock: SplashViewModelType {
```

**Review by mhmdatallaa — COMMENTED** [review]
Great work ya Doaa🌠

File: HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift (hunk 10 above)
Comment (djayyab) [reply HealthyTests/Classes/Modules/Splash/SplashViewControllerTest.swift]: I made it SplashViewModelMock and I am not the one who changed it to SplashViewModelSpy

**Review by mhmdatallaa — APPROVED** [review]

File: HealthyTests/Classes/Modules/Login/Mocks/LoginViewModelMock.swift [hunk 11]
```diff
@@ -59,4 +59,4 @@ final class LoginViewModelMock: LoginViewModelType {
     func performSignInWithFacebook() {
         performSignInWithFacebookCallCount += 1
     }
-}
+ }
```
Comment (ahmdmhasn) [inline HealthyTests/Classes/Modules/Login/Mocks/LoginViewModelMock.swift]:
```suggestion
}
```

File: Healthy/Classes/Modules/Login/LoginViewController.swift [hunk 12]
```diff
@@ -103,10 +103,10 @@ private extension LoginViewController {
     }
 
     func configureViewModelOutputs() {
-        bindLoadingIndicator()
-        bindErrorMessage()
+//        bindLoadingIndicator()
+//        bindErrorMessage()
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Login/LoginViewController.swift]:
```suggestion
        bindLoadingIndicator()
        bindErrorMessage()
```

File: Healthy/Classes/Modules/Login/LoginViewController.swift [hunk 13]
```diff
@@ -103,10 +103,10 @@ private extension LoginViewController {
...
         bindButtonState()
-        bindLoginStatus()
+//        bindLoginStatus()
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Login/LoginViewController.swift]:
```suggestion
        bindLoginStatus()
```

File: Healthy.xcodeproj/project.pbxproj [hunk 14]
```diff
@@ -930,7 +941,9 @@
 				6DFEC6252A2EA5770090B2E2 /* LoginViewModelMock.swift in Sources */,
 				B29ABA5C2A3469CA00171A0C /* MainTapBarControllerTest.swift in Sources */,
 				02279EA72A3373C400C607AD /* CreateAccountViewModelTests.swift in Sources */,
+				C9E6BBE92A38D7EA0073E51A /* SplashViewControllerTest.swift in Sources */,
 				206FF79A2A3A32DD00130CA4 /* CreateViewModelMock.swift in Sources */,
+
```
Comment (ahmdmhasn) [inline Healthy.xcodeproj/project.pbxproj]:
```suggestion
```

**Review by ahmdmhasn — APPROVED** [review]
🚀🚀🚀

**Pull request merged**